var/spack/repos/builtin/packages/umap/package.py

```diff
@@ -17,21 +19,54 @@ class Umap(CMakePackage):
 
     tags = ['e4s']
 
+    maintainers = ['egreen77', 'ibpeng', 'mayagokhale']
+
+    test_requires_compiler = True
+
     version('develop', branch='develop')
     version('2.1.0', sha256='dfdc5b717aecdbfbb0da22e8567b9f2ffbc3607000a31122bf7c5ab3b85cecd9')
     version('2.0.0', sha256='85c4bc68e8790393847a84eb54eaf6fc321acade382a399a2679d541b0e34150')
     version('1.0.0', sha256='c746de3fae5bfc5bbf36234d5e888ea45eeba374c26cd8b5a817d0c08e454ed5')
-    version('0.0.4', sha256='bffaa03668c95b608406269cba6543f5e0ba37b04ac08a3fc4593976996bc273')
-    version('0.0.3', sha256='8e80835a85ad69fcd95f963822b1616c782114077d79c350017db4d82871455c')
-    version('0.0.2', sha256='eccc987b414bc568bd33b569ab6e18c328409499f11e65ac5cd5c3e1a8b47509')
-    version('0.0.1', sha256='49020adf55aa3f8f03757373b21ff229d2e8cf4155d54835019cd4745c1291ef')
+
+    # C++11 support is required for umap
+    depends_on('gcc@4.8.5:', type=('test'))
 
     variant('logging', default=False, description='Build with logging enabled.')
-    variant('tests', default=False, description='Build test programs.')
+    variant('tests', default=False, description='Build examples/tests.')
+    variant('stats', default=False, description='Print internal stats when closing.')
+
+    @run_before("cmake")
```

Comment:
Hey @egreen77 nice job working to integrate the tests into this package! Instead of writing this custom platform check, I'd suggest we update this to a `conflicts` statement. I just recently did [this over on the flux-core package](https://github.com/spack/spack/blob/38d4fd7711a868e0441e1b000a07a7ff94407825/var/spack/repos/builtin/packages/flux-core/package.py#L114) so I'm happy to answer any questions if you run into issues. 


The main benefit to using a conflicts statement is that this'll be evaluated during the concretization of the package and so users won't get all of the way though a long dependency build before running into a, "cannot build on non-linux machines" message.